src/Nethermind/Nethermind.BeaconNode.Peering/PeerDetails.cs

```diff
@@ -14,37 +14,32 @@
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
 
-using Nethermind.Core2.Types;
+using System.Diagnostics;
+using Nethermind.Core2.P2p;
 
 namespace Nethermind.BeaconNode.Peering
 {
-    public class PeerSyncStatus
+    public class PeerDetails
     {
```

Comment:
One of the biggest improvements in Eth1 was a separation of Peer and Session classes.
Peer has now OutSession and InSession - that helps to solve the problem of how to track sessions if you are conencted both ways for a moment and how to drop one f the sessions while keeping reference to the other. Also some pees will misbehave and connect to us many times. It is super important to hold only one peer object for the given peer id - otherwise hell breaks loose.

Reply:
Good idea. Should be a simple change for now; I 'll have a look at the Eth1 code, then split the two objects, even if the current behaviour is very simple, having an architecture design based on the experience from Eth1 will make future handling of these situations easier.

Reply:
I've separated out session and peer, although currently it is very simple, and just has one... but at least is the right structure for the future.